dashboard/app/controllers/api/v1/pd/facilitator_program_registrations_controller.rb

```diff
@@ -0,0 +1,31 @@
+class Api::V1::Pd::FacilitatorProgramRegistrationsController < ApplicationController
+  authorize_resource class: 'Pd::FacilitatorProgramRegistration', only: :create
+
+  def create
+    form_data_hash = params.try(:[], :form_data) || {}
+    form_data_json = form_data_hash.to_unsafe_h.to_json.strip_utf8mb4
+
+    teachercon = params.try(:[], :teachercon)
+
+    facilitator_program_registration = ::Pd::FacilitatorProgramRegistration.new(
+      user: current_user,
+      teachercon: teachercon,
+      form_data: form_data_json
+    )
+
+    begin
+      facilitator_program_registration.save
+    rescue
+      # Fail silently - this happens in the event of multiple submission
+      CDO.log.warn "Duplicate pd facilitator program registration submission for user #{current_user.id} - ignoring the second one"
```

Comment:
(as we discussed offline, repeating here for posterity) note that I am here following [the example set by the teacher applications](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/controllers/api/v1/pd/teacher_applications_controller.rb#L33-L40), and also that it's only possible to view the form in the first place if you have not already made a submission, meaning that you'd have to have opened the form page twice in two separate windows, and submitted first the one then the other. This seems sufficiently edge-case-y to not be worth the time to implement the suggested better solution